### PR TITLE
CORDA-2780 Document FinalityFlow error handling logic and reference flow hospital

### PR DESCRIPTION
**Pull request by josecoll**
https://r3-cev.atlassian.net/browse/CORDA-2780

This PR reverts back some documentation that was deleted with additional improvements, including the Flow Hospital document added to Enterprise (https://github.com/corda/enterprise/pull/1928) but stripped of its Enterprise-only UI visualization section.

### CODE REVIEW

**Comment by josecoll** [comment]
@shamsasari please note the Flow Hospital document has already been reviewed and merged to Enterprise.

File: docs/source/api-flows.rst [hunk 1]
```diff
@@ -613,6 +613,21 @@ flow to receive the transaction:
 ``idOfTxWeSigned`` is an optional parameter used to confirm that we got the right transaction. It comes from using ``SignTransactionFlow``
 which is described below.
 
+**Error handling behaviour**
+
+Once a transaction has been notarised and its input states consumed by the flow initiator (eg. sender), should the participant(s) receiving the
+transaction fail to verify it, or the receiving flow (the finality handler) fails due to some other error, we then have a scenario where not
+all parties have the correct up to date view of the ledger (a condition defined as `eventual consistency <https://en.wikipedia.org/wiki/Eventual_consistency>`_
+in distributed systems terminology). To recover from this scenario, the receivers finality handler will automatically be sent to the
```
Comment (gendal) [inline docs/source/api-flows.rst]: strictly speaking, Corda is only eventually consistent even on the happy path!  How about: "a condition where eventual consistency between participants takes longer than is normally the case under Corda's eventual consistency model [link] "?

also: receiver's

File: docs/source/api-flows.rst [hunk 2]
```diff
@@ -613,6 +613,21 @@ flow to receive the transaction:
...
+:doc:`node-flow-hospital` where it's suspended and retried from its last checkpoint upon node restart, or according to other conditional retry rules
+explained in :ref:`flow hospital runtime behaviour <flow-hospital-runtime>`. This gives the node operator the opportunity to recover from the error.
+Until the issue is resolved the node will continue to retry the flow on each startup. Upon successful completion by the receivers finality flow,
```
Comment (gendal) [inline docs/source/api-flows.rst]: receiver's

File: docs/source/node-flow-hospital.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
+Flow Hospital
+=============
+
+Overview
+--------
+
+The **flow hospital** refers to a built-in node service that manages flows that have encountered an error.
+
+This service is responsible for recording, tracking, diagnosis, recovery and retry. It determines whether errored flows should be retried
```
Comment (gendal) [inline docs/source/node-flow-hospital.rst]: diagnosing, recovering and retrying (ie make it consistent so everything is an 'ing' ! )

File: docs/source/node-flow-hospital.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+from their previous checkpoints or have their errors propagate. Flows may be recoverable under certain scenarios (eg. manual intervention
+may be required to install a missing contract JAR version). For a given errored flow, the flow hospital service determines the next course of
+action towards recovery and retry.
+
+.. note:: The flow hospital will never terminate a flow, but will propagate its error back to the state machine, and ultimately, end user code to handle.
```
Comment (gendal) [inline docs/source/node-flow-hospital.rst]: ... if it ultimately proves impossible to resolve automatically?

File: docs/source/node-flow-hospital.rst (hunk 3 above)
Comment (josecoll) [reply docs/source/node-flow-hospital.rst]: I'll change it but it sounds too 'ing'y

**Pull request merged**